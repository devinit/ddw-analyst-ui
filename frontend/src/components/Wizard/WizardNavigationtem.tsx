import classNames from 'classnames';
import React, { FunctionComponent } from 'react';
import { Nav, NavLinkProps } from 'react-bootstrap';

interface ComponentProps extends NavLinkProps {
  className?: string;
}

const WizardNavigationItem: FunctionComponent<ComponentProps> = ({
  children,
  className,
  ...props
}) => {
  return (
    <Nav.Item as="li" className={classNames(className)}>
      <Nav.Link {...props}>{children}</Nav.Link>
    </Nav.Item>
  );
};

export { WizardNavigationItem };
