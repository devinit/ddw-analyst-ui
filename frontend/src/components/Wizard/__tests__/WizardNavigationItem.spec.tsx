/**
 * @jest-environment jsdom
 */
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { WizardNavigationItem } from '../WizardNavigationtem';

describe('WizardNavigationItem', () => {
  test('renders the default structure correctly', () => {
    const renderer = TestRenderer.create(<WizardNavigationItem />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders with children', () => {
    const renderer = TestRenderer.create(
      <WizardNavigationItem>
        <div>My Child</div>
      </WizardNavigationItem>,
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
