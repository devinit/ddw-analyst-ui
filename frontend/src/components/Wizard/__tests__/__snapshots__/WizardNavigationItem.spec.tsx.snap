// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardNavigationItem renders the default structure correctly 1`] = `
.c0.nav-link {
  display: block;
}

<li
  className="nav-item"
>
  <a
    className="c0 nav-link"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  />
</li>
`;

exports[`WizardNavigationItem renders with children 1`] = `
.c0.nav-link {
  display: block;
}

<li
  className="nav-item"
>
  <a
    className="c0 nav-link"
    href="#"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
  >
    <div>
      My Child
    </div>
  </a>
</li>
`;
